cinema/views.py

```diff
@@ -101,8 +95,27 @@ def get_serializer_class(self):
         if self.action == "retrieve":
             return MovieDetailSerializer
 
+        if self.action == "upload_image":
+            return MovieImageSerializer
```

Comment:
If you need a specific serializer class for custom action you can pass serializer_class as the argument to the action decorator. Read more: https://www.django-rest-framework.org/api-guide/viewsets/#viewset-actions:~:text=The%20decorator%20allows%20you%20to%20override%20any%20viewset%2Dlevel%20configuration%20such%20as%20permission_classes%2C%20serializer_class%2C%20filter_backends...%3A

Reply:
So, if you pass serializer_class to the action decorator, you don't need a specific `if` for this action inside get_serializer_class.